sql/importer: TestImportAvro failed

sql/importer.TestImportAvro [failed](https://tanzanite.cluster.engflow.com/invocations/default/ceca0c9a-cbe7-48cc-8d93-54c84e7e57b1?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on release-24.3.3-rc @ [7dab2bb943152658ca94aeae83c450742ec361a5](https://github.com/cockroachdb/cockroach/commits/7dab2bb943152658ca94aeae83c450742ec361a5):


```
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/strict-import-errors-missing-fields
    --- FAIL: TestImportAvro/strict-import-errors-missing-fields (0.00s)
=== RUN   TestImportAvro/fail-import-expect-ocf-got-json
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/fail-import-expect-ocf-got-json
    --- FAIL: TestImportAvro/fail-import-expect-ocf-got-json (0.00s)
=== RUN   TestImportAvro/import-avro-fragments
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/import-avro-fragments
    --- FAIL: TestImportAvro/import-avro-fragments (0.00s)
=== RUN   TestImportAvro/import-json-pretty-printed-records
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/import-json-pretty-printed-records
    --- FAIL: TestImportAvro/import-json-pretty-printed-records (0.00s)
=== RUN   TestImportAvro/import-json-records-into-table-ignores-extra-fields
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/import-json-records-into-table-ignores-extra-fields
    --- FAIL: TestImportAvro/import-json-records-into-table-ignores-extra-fields (0.00s)
=== RUN   TestImportAvro/relaxed-import-sets-missing-fields
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/relaxed-import-sets-missing-fields
    --- FAIL: TestImportAvro/relaxed-import-sets-missing-fields (0.00s)
=== RUN   TestImportAvro/strict-import-errors-extra-fields
    import_stmt_test.go:6702: 
        	Error Trace:	pkg/sql/importer/import_stmt_test.go:6702
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestImportAvro/strict-import-errors-extra-fields
    --- FAIL: TestImportAvro/strict-import-errors-extra-fields (0.00s)
=== RUN   TestImportAvro/user-defined-schemas
    import_stmt_test.go:6728: error executing query="CREATE SCHEMA myschema" args=[]: sql: database is closed
    --- FAIL: TestImportAvro/user-defined-schemas (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportAvro.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -2019,6 +2019,8 @@ func TestFailedImportGC(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	skip.UnderRace(t)
+
 	const nodes = 3
 
 	var forceFailure bool
@@ -2759,6 +2761,9 @@ INSERT INTO foo VALUES (1);
 func TestImportObjectLevelRBAC(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t)
+
 	const nodes = 3
 
 	ctx := context.Background()
@@ -3838,9 +3843,7 @@ func TestImportIntoCSV(t *testing.T) {
 
 func benchUserUpload(b *testing.B, uploadBaseURI string) {
 	defer log.Scope(b).Close(b)
-	const (
-		nodes = 3
-	)
+	const nodes = 3
 	ctx := context.Background()
 	baseDir, cleanup := testutils.TempDir(b)
 	defer cleanup()
@@ -4109,6 +4112,7 @@ func TestImportDefault(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t, "takes >1min under race")
+
 	const nodes = 3
 	numFiles := nodes + 2
 	rowsPerFile := 1000
@@ -4534,6 +4538,7 @@ func TestImportDefaultNextVal(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	defer setImportReaderParallelism(1)()
+
 	skip.UnderRace(t, "test hits a timeout before a successful run")
 
 	const nodes = 3
@@ -5420,9 +5425,7 @@ func TestImportMysql(t *testing.T) {
 
 	skip.UnderRace(t)
 
-	const (
-		nodes = 3
-	)
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t)
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -5550,9 +5553,9 @@ func TestImportIntoMysql(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t)
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -5577,9 +5580,9 @@ func TestImportDelimited(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t, "mysqlout")
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -5667,9 +5670,9 @@ func TestImportPgCopy(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t, "pgcopy")
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -5753,9 +5756,9 @@ func TestImportPgDump(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t)
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -6503,9 +6506,9 @@ func TestImportCockroachDump(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t)
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -6602,9 +6605,9 @@ func TestImportAvro(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t, "avro")
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -6929,9 +6932,9 @@ func TestDetachedImport(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t, "avro")
 	args := base.TestServerArgs{ExternalIODir: baseDir}
@@ -7029,12 +7032,11 @@ func TestImportRowErrorLargeRows(t *testing.T) {
 func TestImportJobEventLogging(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.ScopeWithoutShowLogs(t).Close(t)
-
 	defer jobs.TestingSetProgressThresholds()()
 
-	const (
-		nodes = 3
-	)
+	skip.UnderRace(t)
+
+	const nodes = 3
 	ctx := context.Background()
 	baseDir := datapathutils.TestDataPath(t, "avro")
 	args := base.TestServerArgs{ExternalIODir: baseDir}

